pkg/component/observability/monitoring/prometheus/prometheus.go

```diff
@@ -136,6 +139,59 @@ func (p *prometheus) prometheus(takeOverOldPV bool, cortexConfigMap *corev1.Conf
 		obj.Spec.RemoteWrite = append(obj.Spec.RemoteWrite, spec)
 	}
 
+	// TODO(rfranzke): Remove this if-block after v1.100 has been released.
+	// For backwards-compatibility with extensions, we still have to mount the shoot CA and access token to the previous
+	// paths in the pod so that their scrape config still works.
+	if p.values.Name == "shoot" && p.values.Ingress.SecretsManager != nil {
+		caSecret, found := p.values.Ingress.SecretsManager.Get(v1beta1constants.SecretNameCACluster)
+		if !found {
+			return nil, fmt.Errorf("secret %q not found", v1beta1constants.SecretNameCACluster)
+		}
+		etcdCASecret, found := p.values.Ingress.SecretsManager.Get(v1beta1constants.SecretNameCAETCD)
+		if !found {
+			return nil, fmt.Errorf("secret %q not found", v1beta1constants.SecretNameCAETCD)
+		}
+		etcdClientSecret, found := p.values.Ingress.SecretsManager.Get("etcd-client")
+		if !found {
+			return nil, fmt.Errorf("secret %q not found", "etcd-client")
+		}
+
+		obj.Spec.Volumes = append(obj.Spec.Volumes,
+			corev1.Volume{
+				Name: "shoot-ca",
+				VolumeSource: corev1.VolumeSource{Projected: &corev1.ProjectedVolumeSource{Sources: []corev1.VolumeProjection{{
+					Secret: &corev1.SecretProjection{
+						LocalObjectReference: corev1.LocalObjectReference{Name: caSecret.Name},
+						// For backwards-compatibility, we make the CA bundle available under both ca.crt and bundle.crt keys.
+						Items: []corev1.KeyToPath{
+							{Key: "bundle.crt", Path: "bundle.crt"},
+							{Key: "bundle.crt", Path: "ca.crt"},
+						},
+						Optional: ptr.To(false),
+					},
+				}}}},
+			},
+			corev1.Volume{
+				Name:         "shoot-access",
+				VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: shootprometheus.AccessSecretName}},
+			},
+			corev1.Volume{
+				Name:         "ca-etcd",
+				VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: etcdCASecret.Name}},
+			},
+			corev1.Volume{
+				Name:         "etcd-client-tls",
+				VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: etcdClientSecret.Name}},
+			},
+		)
+		obj.Spec.VolumeMounts = append(obj.Spec.VolumeMounts,
+			corev1.VolumeMount{Name: "shoot-ca", MountPath: "/etc/prometheus/seed"},
+			corev1.VolumeMount{Name: "shoot-access", MountPath: "/var/run/secrets/gardener.cloud/shoot/token"},
+			corev1.VolumeMount{Name: "ca-etcd", MountPath: "/srv/kubernetes/etcd/ca"},
+			corev1.VolumeMount{Name: "etcd-client-tls", MountPath: "/srv/kubernetes/etcd/client"},
+		)
```

Comment:
Probably not relevant as this section is supposed to go away soon anyway, but the volume names could be constants so that the constant can be directly referenced in the volume mounts, e.g.

```
const shootCAVolumeName = "shoot-ca"
...
			corev1.Volume{
				Name: shootCAVolumeName,
...
			corev1.VolumeMount{Name: shootCAVolumeName, MountPath: "/etc/prometheus/seed"},
```